src/cl/ucc_cl.c

```diff
@@ -99,3 +94,37 @@ ucc_status_t ucc_parse_cls_string(const char *cls_str,
     return UCC_OK;
 }
 
+UCC_CLASS_INIT_FUNC(ucc_cl_context_t, ucc_cl_lib_t *cl_lib)
+{
+    self->super.lib = &cl_lib->super;
+    return UCC_OK;
+}
+
+UCC_CLASS_CLEANUP_FUNC(ucc_cl_context_t)
+{
+}
+
+UCC_CLASS_DEFINE(ucc_cl_context_t, void);
+
+ucc_status_t ucc_cl_context_config_read(ucc_cl_lib_t *cl_lib,
```

Comment:
Same comment as before - maybe I'm missing something, but isn't it easier to have just one config read function?

Reply:
Replying to both comments: essentially there is one common function - ucc_base_config_read  which reads config using ucs. CL/TL versions are just slightly different: they take config tables from different structs and take slightly different args (as defined by tl and cl iface). So, essentially the functions in CL/TL are convenience implementations with more clean implementation (specific to CL or TL)